### ISSUE TITLE
Add missing bzip2 dependency to fontconfig

### ISSUE THREAD

**Issue by kserradell** [LLNL/spack#2589]
Hello, it's me again.

When trying `spack install fontconfig`, I have this error:

`configure: error: Package requirements (freetype2) were not met:`
`Package 'bzip2', required by 'freetype2', not found`

I've tried to add `depends_on('bzip2')` in `var/spack/repos/builtin/packages/fontconfig/package.py` but it still fails.

Any issue?

**Comment by adamjstewart** [comment]
@kserradell Way ahead of you. See #2536. If you update to the latest develop and rebuild all of those dependencies, you should be good to go.

**Comment by kserradell** [comment]
@adamjstewart I'm sorry but even cloning the spack repo in develop branch and starting from scratch, I'm getting this error.

**Comment by adamjstewart** [comment]
Hmm, let me do some debugging.

**Comment by kserradell** [comment]
Thanks. Let me know if you need more info on our environment.

**Comment by adamjstewart** [comment]
I'm unable to reproduce this problem. Which elicits a few questions. I tried `sudo dnf remove bzip2` just to see what would happen if bzip2 wasn't installed on your system, but I can't even install fontconfig because bzip2 is required to decompress the xz tarball. Do you really not have bzip2 installed?

When you run `spack edit freetype`, does it say `depends_on('bzip2')`? If you are definitely on the latest develop commit then it should. This is the commit that was supposed to fix your issue.

When you are rebuilding, you're definitely uninstalling freetype and trying again right? If you had an old freetype installation lying around, it would still be missing the bzip2 dependency.

Can you run the following command? Figure out where freetype gets installed (let's call it $FREETYPE_ROOT) and run:
```
ldd -r $FREETYPE_ROOT/lib/libfreetype.so
```
Also, I have no idea where this error message is coming from:
```
configure: error: Package requirements (freetype2) were not met:
Package 'bzip2', required by 'freetype2', not found
```
This text isn't in fontconfig's configure script.

**Comment by kserradell** [comment]
Hello @adamjstewart. I'm answering all your questions:

1. I do have bzip2 in our OpenSUSE 13.2 system. `zypper search bzip2` shows `bzip2` package.
2. Yes, `spack edit freetype` shows `depends_on('bzip2')`
3. I've started in two different machines with a git clone so my environment is clean.
4. `ldd libfreetype.so`
`	linux-vdso.so.1 (0x00007ffdb570a000)`
`	libbz2.so.1.0 => /home/Earth/kserrade/soft/spack/opt/spack/linux-opensuse13-x86_64/gcc-4.8/bzip2-1.0.6-dqyqjhzboq5g6dselggzh46t7e556wvc/lib/libbz2.so.1.0 (0x00007fb156c8e000)`
`	libpng16.so.16 => /home/Earth/kserrade/soft/spack/opt/spack/linux-opensuse13-x86_64/gcc-4.8/libpng-1.6.26-xv3u62cfjfufjb57d6ilahzvexvdvuv3/lib64/libpng16.so.16 (0x00007fb156a5c000)`
`	libz.so.1 => /home/Earth/kserrade/soft/spack/opt/spack/linux-opensuse13-x86_64/gcc-4.8/zlib-1.2.8-lkabojhp6chwkk7qotzldiqz2dnub3rc/lib/libz.so.1 (0x00007fb15683f000)`
`	libm.so.6 => /lib64/libm.so.6 (0x00007fb156508000)`
`	libc.so.6 => /lib64/libc.so.6 (0x00007fb156161000)`
`	/lib64/ld-linux-x86-64.so.2 (0x00007fb15713a000)`

5. Yes it's really strange. Entering in `/tmp/kserrade/spack-stage/spack-stage-eelXNB/fontconfig-2.11.1`, taking configure from `config.log` and doing `make`, works fine. 

Don't understand.

**Comment by kserradell** [comment]
@adamjstewart digging in fontconfig configure, in line 14323

`Package requirements (freetype2) were not met:`

By the way, in a MacOS works perfectly...

**Comment by adamjstewart** [comment]
This is all very strange. There's something going on here, but I have no idea what. Maybe someone else can figure it out? In the meantime, I would contact the fontconfig and/or freetype developers and see if they have any advice, or can at least explain where that error message is coming from.

**Comment by kserradell** [comment]
Hello. I'll do it.

**Comment by kserradell** [comment]
@adamjstewart ! I've found the problem! Take a look to e98db3441c45129aaa747e5ccbba5b56a101526d

It now installs in my environment.

**Comment by adamjstewart** [comment]
Interesting. I wonder if adding pkg-config to any of the dependencies would solve the problem. Can you try building the dependencies and see if they look for pkg-config during configure? I'm fine with your solution too.

**Comment by kserradell** [comment]
Adding pkg-config dependency in freetype and libxml2 (font-util already have), also solves the problem.

dafcd5e99516ca40642e61ba316b49194ddc5682

Which should be the solution?

**Comment by adamjstewart** [comment]
Unless someone else has an opinion, I think it would be better to add the pkg-config dependency to those packages. If it is installed on your system, the build will pick it up anyway. This just ensures that it is always present and the correct version is always found. Can you submit a PR with that last commit?

**Comment by adamjstewart** [comment]
I think this can be closed now that #2652 has been merged. Reopen if I'm wrong.
